internal/dnfjson/cache.go

```diff
@@ -186,9 +190,12 @@ func (r *rpmCache) updateCacheDirInfo(path string) {
 	// sort IDs by mtime (oldest first)
 	sort.Slice(repoIDs, sortFunc)
```

Comment:
I think that need to sort the repos in `updateInfo`, otherwise `r.repoRecency` will be only "partially" sorted.

Reply:
right, I guess there are no tests for this as well, I will add one

Reply:
The way it works now is that it keeps a per-distro cache and the recency and size are only relevant within each distro.  I wonder if that's the source of the issue.  It means that the max size becomes `maxSize * n distros`.  But we had a lot more than a handful of GiB right?